homeassistant/helpers/config_validation.py

```diff
@@ -402,3 +402,9 @@ def validator(value):
     [vol.Any(SERVICE_SCHEMA, _SCRIPT_DELAY_SCHEMA, EVENT_SCHEMA,
              CONDITION_SCHEMA)],
 )
+
+
+def string_schema(*args: Sequence) -> Dict:
+    """Return a schema dict of Required values or Optional values."""
+    return {val if isinstance(val, vol.Optional) else vol.Required(val): string
+            for val in args}
```

Comment:
It can cut down you ODB tracker schema like this:

```python
PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
    vol.Required(CONF_CLIENT_ID): cv.string,
    vol.Required(CONF_SECRET): cv.string,
    vol.Required(CONF_USERNAME): cv.string,
    vol.Required(CONF_PASSWORD): cv.string,
    vol.Optional(CONF_DEVICES): vol.All(cv.ensure_list, [cv.string])
})
```

to 

```python
PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
    string_schema(CONF_CLIENT_ID, CONF_SECRET, CONF_USERNAME, CONF_PASSWORD)
).extend({
    vol.Optional(CONF_DEVICES): vol.All(cv.ensure_list, [cv.string])
})
```